tests/lib/test_azure.py

```diff
@@ -64,3 +51,178 @@ def test_client_copying_override_options(client: Client) -> None:
         api_version="2022-05-01",
     )
     assert copied._custom_query == {"api-version": "2022-05-01"}
+
+
+@pytest.mark.parametrize(
+    "client,base_url,api_path,json_data,expected",
+    [
+        (
+            AzureOpenAI(
+                api_version="2024-02-01",
+                api_key="example API key",
+                azure_endpoint="https://example-resource.azure.openai.com",
+            ),
+            "https://example-resource.azure.openai.com/openai/",
+            "/chat/completions",
+            {"model": "my-deployment"},
+            "https://example-resource.azure.openai.com/openai/deployments/my-deployment/chat/completions?api-version=2024-02-01"
+        ),
+        (
+            AzureOpenAI(
+                api_version="2024-02-01",
+                api_key="example API key",
+                azure_endpoint="https://example-resource.azure.openai.com",
+            ),
+            "https://example-resource.azure.openai.com/openai/",
+            "/models",
```

Comment:
super nitpick: assistants would be more interesting I reckon

Reply:
Good point, I added tests for chat completions, assistants, and models since they each represent a different scenario with respect to deployments